lib/reporter.js

```diff
@@ -25,190 +30,133 @@ class AllureReporter extends events.EventEmitter {
-
-                        if (this.preTestOutput && this.preTestOutput.length) {
-                            this.startStep('before first test', this.preTestOutput[0].payload.time.getTime())
-                            this.addOutput(this.preTestOutput)
-                            this.flushCommandOutput()
-                            const endTime = this.preTestOutput[(this.preTestOutput.length - 1)].payload.time.getTime()
-                            this.endStep('passed', endTime)
-                            this.reconcileSteps(endTime)
-                            delete this.preTestOutput
-                        }
-
-                        this.addOutput(test.output)
-                        this.reconcileSteps()
-
-                        if (this.commandOutput.length) {
-                            this.startStep('after last command')
-                            this.flushCommandOutput()
-                            this.endStep('passed')
-                        }
-
-                        if (test.state === 'pass') {
-                            this.allure.endCase('passed')
-                        } else if (test.error && test.error.type === 'AssertionError') {
-                            this.allure.endCase('failed', test.error)
-                        } else {
-                            this.allure.endCase('broken', test.error)
-                        }
-                    }
-                    this.allure.endSuite(spec.suites[suiteName].end && spec.suites[suiteName].end.getTime())
-                    console.log(`Wrote Allure report for ${suiteName} to [${this.outputDir}].`)
-                }
-            }
-        }
-    }
+        this.on('suite:start', (suite) => {
+            ++this.level
+            this.pushParent(suite.title)
+            // the root suite groups together in the dashboard
+            var title = this.parents[0]
+            this.print('suite:start >', title)
+            this.allure.startSuite(title, this.getTime())
+        })
 
-    padTime (num, size = 2) {
-        if (num >= Math.pow(10, size - 1)) return num
-        return (Math.pow(10, size) + num).toString().substring(1)
-    }
+        this.on('suite:end', (suite) => {
+            --this.level
+            this.popParent()
+            var title = this.parents.length ? this.parents[0] : this.getTitle(suite)
+            this.print('test:end >', title)
+            this.allure.endSuite(this.getTime())
+            console.log(`Wrote Allure report for ${title} to [${this.outputDir}].`)
+        })
 
-    formatTime (time) {
-        return `${this.padTime(time.getHours())}:${this.padTime(time.getMinutes())}:${this.padTime(time.getSeconds())}.${this.padTime(time.getMilliseconds(), 3)}`
-    }
+        this.on('test:start', (test) => {
+            var title = this.getTitle(test)
+            this.pushParent(title)
+            this.print('test:start >', title)
+            this.allure.startCase(title, this.getTime())
```

Comment:
If you do not pass the time, Allure uses `Date.now()` by default. You can get rid of `getTime()` method at all, when you are doing real-time reporting.